modules/nf-scil/bundle/labelmap/main.nf

```diff
@@ -0,0 +1,70 @@
+
+process BUNDLE_LABELMAP {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
+        'scilus/scilus:2.0.2' }"
+
+    input:
+        tuple val(meta), path(bundles), path(centroids)
+
+    output:
+    tuple val(meta), path("*_labels.nii.gz")    , emit: labels
+    tuple val(meta), path("*_labels.trk")       , emit: labels_trk
+    tuple val(meta), path("*_distances.nii.gz") , emit: distances
+    tuple val(meta), path("*_distances.trk")    , emit: distances_trk
+    path "versions.yml"                         , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def nb_points = task.ext.nb_points ? "--nb_pts ${task.ext.nb_points} ": ""
+    def colormap = task.ext.colormap ? "--colormap ${task.ext.colormap} ": ""
+    def new_labelling = task.ext.new_labelling ? "--new_labelling ": ""
+
+    """
+    bundles=(${bundles.join(" ")})
+    centroids=(${centroids.join(" ")})
+
+    for index in \${!bundles[@]};
+        do ext=\${bundles[index]#*.}
+        bname=\$(basename \${bundles[index]} .\${ext})
+
+        if [[ -f \${centroids[index]} ]]; then
```

Comment:
Remove the `if`, I would prefer the process to fail if the centroids list doesn't have the same size.